src/Symfony/Component/HttpFoundation/Session/Storage/Handler/Psr6SessionHandler.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Session\Storage\Handler;
+
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * Session handler that supports a PSR6 cache implementation.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Ahmed TAILOULOUTE <ahmed.tailouloute@gmail.com>
+ */
+class Psr6SessionHandler extends AbstractSessionHandler
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cache;
+
+    /**
+     * @var int Time to live in seconds
+     */
+    private $ttl;
+
+    /**
+     * @var string Key prefix for shared environments
+     */
+    private $prefix;
+
+    /**
+     * List of available options:
+     *  * prefix: The prefix to use for the cache keys in order to avoid collision
+     *  * ttl: The time to live in seconds.
+     *
+     * @param CacheItemPoolInterface $cache   A Cache instance
+     * @param array                  $options An associative array of cache options
+     */
+    public function __construct(CacheItemPoolInterface $cache, array $options = [])
+    {
+        $this->cache = $cache;
+
+        if ($diff = array_diff(array_keys($options), ['prefix', 'ttl'])) {
+            throw new \InvalidArgumentException(sprintf('The following options are not supported "%s"', implode(', ', $diff)));
```

Comment:
`sprintf('The following options are not supported by %s: "%s".', static::class, implode('", "', $diff))`